src/TADA_GetATTAINS.R

```diff
@@ -75,21 +63,93 @@ TADA_GetATTAINS <- function(data){
       urltools::param_set(key = "returnExtentOnly", value = "false") %>%
       urltools::param_set(key = "featureEncoding", value = "esriDefault")
     
-    # grab the geospatial data!
-    nearby_catchments <- geojsonsf::geojson_sf(query) %>%
-      # remove unnecessary columns:
-      dplyr::select(-c(OBJECTID, GLOBALID)) %>%
-      .[TADA_DataRetrieval_data, ] %>%
-      dplyr::rename_with(~ paste0("ATTAINS.", .), dplyr::everything()) %>%
-      dplyr::distinct(.keep_all = TRUE)
-    
-    # join TADA sf features to the ATTAINS catchment feature(s) they land on:
-    TADA_with_ATTAINS <- TADA_DataRetrieval_data %>%
-      sf::st_join(., nearby_catchments) %>%
-      sf::st_drop_geometry()
-    
-    return(TADA_with_ATTAINS)
+    # grab the ATTAINS catchments within our WQP bbox:
+    # (Wrapped with "try" because it is possible that no ATTAINS data exists in the bbox.)
+    try(nearby_catchments <- geojsonsf::geojson_sf(query) %>%
+          # remove unnecessary columns:
+          dplyr::select(-c(OBJECTID, GLOBALID)) %>%
+          # select only catchments that have WQP observations in them:
+          .[TADA_DataRetrieval_data, ] %>%
+          # prepend "ATTAINS" to ATTAINS data
+          dplyr::rename_with(~ paste0("ATTAINS.", .), dplyr::everything()) %>%
+          # get rid of dupes (as a precaution)
+          dplyr::distinct(.keep_all = TRUE),
+        silent = TRUE)
+    # if not ATTAINS data return original data frame:
+    if(is.null(nearby_catchments)){
+      print("There are no ATTAINS features associated with these WQP observations. Returning original dataframe.")
+      return(data)
```

Comment:
thought: Would it be better to have this produce an error? When reading this made me wonder if someone trying to iterate with this could inadvertently try using the original dataset, which wouldn't have the same structure as what they'd anticipate as output.

Reply:
I agree - this was something done in another TADA function that I copied. But your suggestion makes more sense! OR - I could add those ATTAINS columns and they are just empty...

Reply:
Ooooh yep that seems viable too!